tests/query/test_matcher.py

```diff
@@ -0,0 +1,143 @@
+from typing import Optional
+
+import pytest
+
+from snuba.query.expressions import Column as ColumnExpr
+from snuba.query.expressions import Expression, FunctionCall
+from snuba.query.matcher import (
+    AnyExpression,
+    AnyString,
+    Column,
+    Function,
+    MatchResult,
+    Or,
+    Param,
+    Pattern,
+    String,
+)
+
+test_cases = [
+    (
+        Column(None, String("test_col"), String("table")),
+        ColumnExpr("alias_we_don't_care_of", "test_col", "table"),
+        MatchResult(),
+    ),  # Simple single node column that matches
```

Comment:
This is more of a personal opinion, but it doesn't seem to me that parametrization is all that helpful here. In the test report, this becomes `tests/query/test_matcher.py::test_base_expression[pattern0-expression0-expected_result0]` which is a lot less informative than `tests/query/test_matcher.py::test_simple_single_node_column_match` or something similar.

Reply:
I gave names to the tests now they look like this:
```
(snuba) ~/snuba  (feat/expr_matcher *)$ SNUBA_SETTINGS=test py.test -vv tests/query/test_matcher.py
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 3.7.4, pytest-5.2.4, py-1.5.3, pluggy-0.13.0 -- /Users/filippopacifici/.virtualenvs/snuba/bin/python3.7
cachedir: .pytest_cache
rootdir: /Users/filippopacifici/snuba, inifile: setup.cfg
plugins: xdist-1.30.0, cov-2.5.1, forked-1.1.3
collected 25 items                                                                                                                                                                                         

tests/query/test_matcher.py::test_base_expression[Single node match-pattern0-expression0-expected_result0] PASSED                                                                                    [  4%]
tests/query/test_matcher.py::test_base_expression[Single node no match-pattern1-expression1-None] PASSED                                                                                             [  8%]
tests/query/test_matcher.py::test_base_expression[Matches a None table name-pattern2-expression2-expected_result2] PASSED                                                                            [ 12%]
tests/query/test_matcher.py::test_base_expression[Matches None as table name-pattern3-expression3-expected_result3] PASSED                                                                           [ 16%]
tests/query/test_matcher.py::test_base_expression[Not matching a non None table-pattern4-expression4-None] PASSED                                                                                    [ 20%]
tests/query/test_matcher.py::test_base_expression[Matches a column with all fields-pattern5-expression5-expected_result5] PASSED                                                                     [ 24%]
tests/query/test_matcher.py::test_base_expression[Match anything-pattern6-expression6-expected_result6] PASSED                                                                                       [ 28%]
tests/query/test_matcher.py::test_base_expression[Match a string through Any(str)-pattern7-expression7-expected_result7] PASSED                                                                      [ 32%]
tests/query/test_matcher.py::test_base_expression[Match a None string through Any-pattern8-expression8-expected_result8] PASSED                                                                      [ 36%]
tests/query/test_matcher.py::test_base_expression[Do not Match a None string through Any-pattern9-expression9-None] PASSED                                                                           [ 40%]
tests/query/test_matcher.py::test_base_expression[Match any expression of Column type-pattern10-expression10-expected_result10] PASSED                                                               [ 44%]
tests/query/test_matcher.py::test_base_expression[Match any expression of Column type within function-pattern11-expression11-expected_result11] PASSED                                               [ 48%]
tests/query/test_matcher.py::test_base_expression[Does not match any Column-pattern12-expression12-None] PASSED                                                                                      [ 52%]
tests/query/test_matcher.py::test_base_expression[Union of two patterns - match-pattern13-expression13-expected_result13] PASSED                                                                     [ 56%]
tests/query/test_matcher.py::test_base_expression[Union of two patterns - no match-pattern14-expression14-None] PASSED                                                                               [ 60%]
tests/query/test_matcher.py::test_base_expression[Or within a Param-pattern15-expression15-expected_result15] PASSED                                                                                 [ 64%]
tests/query/test_matcher.py::test_base_expression[Match String Literal-pattern16-expression16-expected_result16] PASSED                                                                              [ 68%]
tests/query/test_matcher.py::test_base_expression[Match any string as Literal-pattern17-expression17-expected_result17] PASSED                                                                       [ 72%]
tests/query/test_matcher.py::test_base_expression[Does not match an int as Literal-pattern18-expression18-None] PASSED                                                                               [ 76%]
tests/query/test_matcher.py::test_base_expression[returns the columns in any function-pattern19-expression19-expected_result19] PASSED                                                               [ 80%]
tests/query/test_matcher.py::test_base_expression[matches a function with optional params-pattern20-expression20-expected_result20] PASSED                                                           [ 84%]
tests/query/test_matcher.py::test_base_expression[dows not match even with optionals-pattern21-expression21-None] PASSED                                                                             [ 88%]
tests/query/test_matcher.py::test_base_expression[nested parameters no match-pattern22-expression22-None] PASSED                                                                                     [ 92%]
tests/query/test_matcher.py::test_base_expression[complex structure matches-pattern23-expression23-expected_result23] PASSED                                                                         [ 96%]
tests/query/test_matcher.py::test_accessors PASSED                                                         
```